crates/holochain_state/src/validation_receipts.rs

```diff
@@ -69,26 +71,27 @@ impl ValidationReceipt {
                 async move { validator.sign(&keystore, this).await }
             })
             .collect::<Vec<_>>();
-        let signatures = futures::stream::iter(futures)
-            .buffer_unordered(10)
-            .collect::<Vec<_>>()
-            .await;
-        let signatures = if signatures.iter().any(Result::is_ok) {
-            signatures.into_iter().filter_map(Result::ok).collect()
-        } else {
-            return Err(signatures
-                .into_iter()
-                .filter_map(Result::err)
-                .next()
-                .expect("Must contain at least one error if there is no success"));
-        };
+        let stream = futures::stream::iter(futures);
+        let signatures = try_stream_of_results(stream).await?;
+        if signatures.is_empty() {
+            return Ok(None);
+        }
         Ok(Some(SignedValidationReceipt {
             receipt: self,
             validators_signatures: signatures,
         }))
     }
 }
 
+/// Try to collect a stream of futures that return results into a vec.
+async fn try_stream_of_results<T, U, E>(stream: U) -> Result<Vec<T>, E>
+where
+    U: Stream,
+    <U as Stream>::Item: futures::Future<Output = Result<T, E>>,
+{
+    stream.buffer_unordered(10).map(|r| r).try_collect().await
```

Comment:
what's the point of `.map(|r| r)`?

Reply:
It's a futures map, it means you go from `Future<Output=T>` to `T`